### PR TITLE
2022 CSS chapter

### LINKED ISSUE #2879: CSS 2022
# CSS 2022

CSS illustration

If you're interested in contributing to the CSS chapter of the 2022 Web Almanac, please reply to this issue and indicate which role or roles best fit your interest and availability: [author](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide), [reviewer](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide), [analyst](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide), and/or [editor](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide).

## Content team

| Lead | Authors | Reviewers | Analysts | Editors | Coordinator |
| ---- | ------- | --------- | -------- | ------- | ----------- |
| @rachelandrew | @rachelandrew | @j9t @meyerweb @svgeesus | @rviscomi | @dereknahman | @siakaramalegos  |

<details>
<summary><b>Expand for more information about each role 👀</b></summary>

- The **[content team lead](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Content-Team-Leads'-Guide)** is the chapter owner and responsible for setting the scope of the chapter and managing contributors' day-to-day progress.
- **[Authors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide)** are subject matter experts and lead the content direction for each chapter. Chapters typically have one or two authors. Authors are responsible for planning the outline of the chapter, analyzing stats and trends, and writing the annual report.
- **[Reviewers](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide)** are also subject matter experts and assist authors with technical reviews during the planning, analyzing, and writing phases.
- **[Analysts](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide)** are responsible for researching the stats and trends used throughout the Almanac. Analysts work closely with authors and reviewers during the planning phase to give direction on the types of stats that are possible from the dataset, and during the analyzing/writing phases to ensure that the stats are used correctly.
- **[Editors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide)** are technical writers who have a penchant for both technical and non-technical content correctness. Editors have a mastery of the English language and work closely with authors to help wordsmith content and ensure that everything fits together as a cohesive unit.
- The **[section coordinator](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Section-Leads'-Guide)** is the overall owner for all chapters within a section like "User Experience" or "Page Content" and helps to keep each chapter on schedule.

_Note: The time commitment for each role varies by the chapter's scope and complexity as well as the number of contributors._

For an overview of how the roles work together at each phase of the project, see the [Chapter Lifecycle](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle) doc.
</details>

### [**0. Form the content team**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#0-create-content-team)
+ [x] **May 1**: The content team has at least one author, reviewer, and analyst

### [**1. Plan content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#1-plan-content)
+ [x] **May 15** The content team has completed the chapter outline in the [draft doc][~google-doc]

### [**2. Gather data**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#2-gather-data)
+ [x] **June 1**: Analysts have added all necessary [custom metrics](https://github.com/HTTPArchive/custom-metrics/blob/main/README.md) and drafted a PR ([example](https://github.com/HTTPArchive/almanac.httparchive.org/pull/1087)) to track query progress
- **June 1 - 15**: HTTP Archive runs the June crawl

### [**3. Validate results**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#3-validate-results)
+ [x] **August 1**: Analysts have queried all metrics and saved the output to the [results sheet][~google-sheets]

### [**4. Draft content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#4-draft-content)
+ [x] **September 1**: The content team has written, reviewed, and edited the chapter in the [doc][~google-doc]

### [**5. Publication**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#5-publication)
+ [ ] **September 15**: The completed chapter and all required [metadata](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#metadata-to-add-at-the-top-of-your-chapters) and [figures](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Figures-Guide) are converted to markdown and submitted to [GitHub][~chapter-markdown]
- **September 26**: Target launch date 🚀

## Chapter resources

Refer to these 2022 CSS resources throughout the content creation process:

[📄 Google Docs][~google-doc] for outlining and drafting content
[🔍 SQL files][~sql-files] for committing the queries used during analysis
[📊 Google Sheets][~google-sheets] for saving the results of queries
[📝 Markdown file][~chapter-markdown] for publishing content and managing public metadata
[💬 #web-almanac-css on Slack][~slack-channel] for team coordination

[~google-doc]: https://docs.google.com/document/d/1aKTQht_HeLdskNOeELI_e3rhTev5ldyiSl72Qbm18O0/edit?usp=sharing
[~sql-files]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/sql/2022/css
[~google-sheets]: https://docs.google.com/spreadsheets/d/1OU8ahxC5oYU8VRryQs9BzHToaXcOntVlh6KUHjm15G4/edit?usp=sharing
[~chapter-markdown]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/content/en/2022/css.md
[~slack-channel]: https://join.slack.com/t/httparchive/shared_invite/zt-45sgwmnb-eDEatOhqssqNAKxxOSLAaA

### PR DESCRIPTION
**Pull request by rviscomi**
Fixes #2879 

Staging link: https://20220921t022138-dot-webalmanac.uk.r.appspot.com/en/2022/css

- [x] Converted doc to markdown @rviscomi 
- [x] Figures migrated and screenshots taken @rviscomi 
- [x] Figure descriptions @rviscomi
- [x] Edited @rviscomi 
- [x] Chapter metadata (contributors, author bio, featured stats/quote) @rachelandrew

### CODE REVIEW

File: src/content/en/2022/css.md [hunk 1]
```diff
@@ -6,8 +6,9 @@ description: CSS chapter of the 2022 Web Almanac covering trends, changes, and p
 authors: [rachelandrew]
 reviewers: [j9t, meyerweb, svgeesus]
 analysts: [rviscomi]
-editors: [dereknahman]
+editors: [rviscomi]
```
Comment (rviscomi) [inline src/content/en/2022/css.md:9]: @rachelandrew is this everyone who has meaningfully contributed to the chapter? Could you also suggest any changes needed to sort the contributors in order of biggest contributors first?

File: src/content/en/2022/css.md [hunk 2]
```diff
@@ -6,8 +6,9 @@ description: CSS chapter of the 2022 Web Almanac covering trends, changes, and p
...
 translators: []
+rachelandrew_bio: TODO
```
Comment (rviscomi) [inline src/content/en/2022/css.md:11]: @rachelandrew could you suggest a bio for your self? 2021 for reference: https://almanac.httparchive.org/en/2021/css#authors

File: src/content/en/2022/css.md [hunk 3]
```diff
@@ -6,8 +6,9 @@ description: CSS chapter of the 2022 Web Almanac covering trends, changes, and p
...
 results: https://docs.google.com/spreadsheets/d/1OU8ahxC5oYU8VRryQs9BzHToaXcOntVlh6KUHjm15G4/
 featured_quote: TODO
 featured_stat_1: TODO
```
Comment (rviscomi) [inline src/content/en/2022/css.md:14]: @rachelandrew could you suggest a quote and three of the most interesting stats from this chapter to be featured on the home page? For example: https://almanac.httparchive.org/en/2021/?feat=css#featured-chapter

**Comment by rviscomi** [comment]
@rachelandrew could you suggest updates to the metadata section?

- your bio
- contributor list
- featured stats/quote

**Comment by rviscomi** [comment]
@rachelandrew I've completed my editing pass over the chapter. No substantive changes to the content, mostly grammatical edits, code formatting for property names, and rearranging figures to come before your interpretation of the data. You can see all of my edits in [this commit](https://github.com/HTTPArchive/almanac.httparchive.org/pull/3140/commits/a095eb75a678151660626d5fcb53648c8d302e57) in case you'd like to check it over. I also left a couple of TODOs for you in the markdown, which I'll also flag in comments. These aren't necessarily launch-blocking but good to address if you have time.

File: src/content/en/2022/css.md [hunk 4]
```diff
@@ -19,4 +20,1308 @@ featured_stat_label_3: TODO
...
+    sheets_gid="377488072",
+    sql_file="meta_important_properties.sql"
+  )
+}}
+
+In terms of what `!important` is applied to, the top properties remain unchanged. However, `position` has fallen out of the top ten, to be replaced with `font-size`.
+
+### Selector specificity
+
+{# TODO(author): Can you introduce what specificity is? #}
```
Comment (rviscomi) [inline src/content/en/2022/css.md:152]: @rachelandrew

File: src/content/en/2022/css.md [hunk 5]
```diff
@@ -19,4 +20,1308 @@ featured_stat_label_3: TODO
...
+    caption="Most popular attribute selectors by percent of pages.",
+    description="TODO",
+    chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSPMUe75uC8laUvzfECAPpT9fPoTdZA6FYDULorsUzKVmLmagphzh1UoXRcmyd6a3gxqr6dxIhmJyv5/pubchart?oid=1566442848&format=interactive",
+    height="675",
+    sheets_gid="1504728475",
+    sql_file="top_selector_attributes.sql"
+  )
+}}
+
+{# TODO(author): This subsection should have an intro and interpretation of the data. #}
```
Comment (rviscomi) [inline src/content/en/2022/css.md:247]: @rachelandrew

File: src/content/en/2022/css.md [hunk 6]
```diff
@@ -6,17 +6,1321 @@ description: CSS chapter of the 2022 Web Almanac covering trends, changes, and p
...
 results: https://docs.google.com/spreadsheets/d/1OU8ahxC5oYU8VRryQs9BzHToaXcOntVlh6KUHjm15G4/
-featured_quote: TODO
-featured_stat_1: TODO
-featured_stat_label_1: TODO
-featured_stat_2: TODO
-featured_stat_label_2: TODO
-featured_stat_3: TODO
-featured_stat_label_3: TODO
-unedited: true
+featured_quote: The past few years have seen a flurry of new CSS features. Many of these have taken inspiration from things developers were already doing with JavaScript or in preprocessors, while others provide methods of doing things that were impossible a few years ago. Having new features available is one thing, but are developers actually using them in their production web pages and applications?
```
Comment (rviscomi) [inline src/content/en/2022/css.md:13]: @rachelandrew I've added some featured quote/stats that I think are interesting but feel free to suggest something else.

File: src/content/en/2022/css.md (hunk 1 above)
Comment (rviscomi) [reply src/content/en/2022/css.md]: Took a pass at shuffling contributors

File: src/content/en/2022/css.md (hunk 2 above)
Comment (rviscomi) [reply src/content/en/2022/css.md]: Added a placeholder bio for now

**Comment by rviscomi** [comment]
@rachelandrew I've resolved all of the remaining TODOs so this is in a good state to merge and release for Monday, but please open a new PR with any suggested changes as needed.

File: src/content/en/2022/css.md (hunk 1 above)
Comment (rachelandrew) [reply src/content/en/2022/css.md]: I think that looks ok.

**Pull request merged**